src/canvas/Canvas.ts

```diff
@@ -1489,9 +1489,14 @@ export class Canvas extends SelectableCanvas {
         const prevActiveObjects =
           activeSelection.getObjects() as FabricObject[];
         if (target === activeSelection) {
-          // find target from active objects
+          // find target from active objects to remove or from under active selection to add
           target = this.searchPossibleTargets(
-            prevActiveObjects,
+            [
+              ...this._objects.filter(
+                (object) => !prevActiveObjects.includes(object)
+              ),
+              ...prevActiveObjects,
+            ],
```

Comment:
I m trying to wrap my head around why we need to change the stack for searchPossibleTarget.
I understand you want the activeSelectionObject searched first, but i think that is job for `altSelectionKey`.
Isn'it?
Shouldn't we use the natural stack order here?

Reply:
> Yes this was made possible in findTarget, i think at the end of the function where we would say that if there wasn't a sub target in the activeSelection, but there was eventually a target under it we would take that one

I followed this logic and this comment

https://github.com/fabricjs/fabric.js/blob/26ed225b47288f775e598f719e218a660dd58dd0/src/canvas/SelectableCanvas.ts#L989

Reply:
I guess we can achieve it in a different way.
If `!target` => search the rest of the object on canvas to see if there is an object behind the selection that we can add